test/e2e/ha_master.go

```diff
@@ -75,26 +76,61 @@ func verifyNumberOfMasterReplicas(expected int) {
 	}
 }
 
+func findRegionForZone(zone string) string {
+	output, err := exec.Command("gcloud", "compute", "regions", "list", "--quiet", "--format=[no-heading](name)").CombinedOutput()
+	framework.ExpectNoError(err)
+	regions := strings.Split(string(output), "\n")
+	for _, region := range regions {
+		if strings.HasPrefix(zone, region) {
+			return region
+		}
+	}
+	framework.Failf("Region not found; zone: %s, regions: %v", zone, regions)
+	return "" // unreachable
+}
+
+func findZonesForRegion(region string) []string {
+	output, err := exec.Command("gcloud", "compute", "zones", "list", "--regexp=^"+region+".*",
+		"--quiet", "--format=[no-heading](name)").CombinedOutput()
+	framework.ExpectNoError(err)
+	zones := strings.Split(string(output), "\n")
+	return zones
+}
+
+func removeZoneFromZones(zones []string, zone string) []string {
```

Comment:
Instead of implementing this on your own you can also use `sets.String` implemented in `pkg/util/sets` or use a map instead where you could just delete a key.